### PR TITLE
Turn in place and nav state refactoring

### PR DESCRIPTION
**Pull request by joydeep-b**

### CODE REVIEW

**Comment by joydeep-b** [comment]
I tested it on the robot in AHG2, works fine with autonomous navigation.

File: config/navigation.lua [hunk 1]
```diff
@@ -29,8 +33,10 @@ NavigationParameters = {
   target_dist_tolerance = 0.1;
   target_vel_tolerance = 0.1;
   target_angle_tolerance = 0.05;
+  local_fov = deg2rad(50);
 };
 
 AckermannSampler = {
   max_curvature = 3;
+  clearance_path_clip_fraction = 0.8;
```
Comment (kvsikand) [inline config/navigation.lua:41]: can you explain what this is doing? I think it has to do with the degenerate case of driving straight at an obstacle, but I'm not totally sure

**Review by kvsikand — APPROVED** [review]

**Review by jarrettHoltz — APPROVED** [review]
I'll check for compatibility with my stuff separately and submit a PR if I need to change anything.

File: config/navigation.lua (hunk 1 above)
Comment (joydeep-b) [reply config/navigation.lua:41]: Yes, it was an attempt to address that, but it doesn't actually work. 
Observations beyond that fraction of the path are not used for clearance computation.  Setting it to 1 will have the same effect as before, so I'll just do that for now.

**Pull request merged**